spec/models/miq_request_workflow_spec.rb

```diff
@@ -152,4 +151,30 @@
       expect(workflow.provisioning_tab_list).to eq([{:name => "test3", :description => "test description 3"}])
     end
   end
+
+  context "'allowed_*' methods" do
+    let(:cluster)       { FactoryGirl.create(:ems_cluster, :ems_id => ems.id) }
+    let(:ems)           { FactoryGirl.create(:ext_management_system) }
+    let(:resource_pool) { FactoryGirl.create(:resource_pool, :ems_id => ems.id) }
+
+    it "#allowed_clusters" do
+      FactoryGirl.create(:ems_cluster)
+      allow_any_instance_of(User).to receive(:get_timezone).and_return("UTC")
+      allow(workflow).to receive(:get_source_and_targets).and_return(:ems => ems)
+
+      expect(workflow).to receive(:allowed_ci).with(:cluster, [:respool, :host, :folder], [cluster.id])
+
+      workflow.allowed_clusters
```

Comment:
It tests that only the clusters for this EMS are in the list and that the call to ```Rbac.filtered``` returns the full list for the admin user.